File: Algorithms/p058_Length_of_Last_Word/p058_Length_of_Last_Word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int pos = s.size()-1;
        while(pos>=0&&s[pos]==' ') --pos;
        int end = pos;
        while(pos>=0&&s[pos]!=' ') --pos;
        return end-pos;
    }
};

File: Algorithms/p058_Length_of_Last_Word/p058_Length_of_Last_Word.py
class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        pos = len(s)-1
        while pos>=0 and s[pos]==' ':
            pos -= 1
        end = pos
        while pos>=0 and s[pos]!=' ':
            pos -= 1
        return end-pos
